src/java/org/apache/cassandra/net/SocketFactory.java

```diff
@@ -237,6 +240,30 @@ static String encryptionLogStatement(EncryptionOptions options)
         return "enabled (" + encryptionType + ')';
     }
 
+    static String encryptionLogStatement(Channel channel, EncryptionOptions options)
+    {
+        if (options == null)
+            return "disabled";
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("enabled (factory=");
+        sb.append(SSLFactory.openSslIsAvailable() ? "openssl" : "jdk");
+
+        final ChannelPipeline pipeline = channel.pipeline();
```

Comment:
this doesn't fix the issue

```
public String toString()
     {
         return format("address: (%s), nic: %s, encryption: %s",
-                       bindAddress, FBUtilities.getNetworkInterface(bindAddress.address), SocketFactory.encryptionLogStatement(encryption));
+                       bindAddress, FBUtilities.getNetworkInterface(bindAddress.address), SocketFactory.encryptionLogStatement(null, encryption));
     }
```

The toString method now passes null as the channel, so the channel itself is null

Reply:
Teach me to fix things at the end of the day.  Pushed up a fix. The silver lining is that it made me look a bit more carefully about the message and I think the current one is misleading so the fix improves the `toString() output to be more useful.

I've updated the JIRA with more details to get more visibility on it.